scripts/create-darwin-volume.sh

```diff
@@ -127,42 +164,75 @@ main() {
-                echo "         See https://nixos.org/nix/manual/#sect-macos-installation" >&2
-            else
-                echo "error: refusing to create Nix store volume because the boot volume is" >&2
-                echo "       FileVault encrypted, but encryption-at-rest is not available." >&2
-                echo "       Manually create a volume for the store and re-run this script." >&2
-                echo "       See https://nixos.org/nix/manual/#sect-macos-installation" >&2
-                exit 1
-            fi
+    if [ -n "$create_volume" ]; then
+        echo "Creating a Nix volume..." >&2
+
+        sudo diskutil apfs addVolume "$disk" "$NIX_VOLUME_FS" "$volume" -mountpoint /nix
+        new_uuid="$(volume_uuid "$volume")"
+
+        if [ "$INSTALL_MODE" = "no-daemon" ]; then # exported by caller
+            # TODO: is there a better way to do this?
+            sudo chown $USER:admin /nix
         fi
 
-        sudo diskutil apfs addVolume "$disk" APFS 'Nix Store' -mountpoint /nix
-        volume="Nix Store"
+        if test_filevault_in_use; then
+            # security program's flags won't let us both specify a keychain
+            # and be prompted for a pw to add; two step workaround:
+            # 1. add a blank pw to system keychain
+
+            # system is in some sense less secure than user keychain... (it's
+            # possible to read the password for decrypting the keychain) but
+            # the user keychain appears to be available too late. As far as I
+            # can tell, the file with this password (/var/db/SystemKey) is
+            # inside the FileVault envelope. If that isn't true, it may make
+            # sense to store the password inside the envelope?
+            sudo /usr/bin/security add-generic-password -a "$volume" -s "$new_uuid" -D "$volume encryption password" -j "Added automatically by the Nix installer for use by /Library/LaunchDaemons/org.nixos.darwin-store.plist" "/Library/Keychains/System.keychain"
+            # TODO: decide if we should add `-T /System/Library/CoreServices/APFSUserAgent`
+            # This should let the system seamlessly supply the password for this volume
+            # which in turn means the fstab entry is enough for the system to (eventually)
+            # decrypt and mount the volume we're adding, but I hesitate because I'm not
+            # certain the system _should_ transparently failover if the LaunchDaemon is
+            # broken for some reason? Without supplying this flag, the system will instead
+            # start prompting them to allow APFSUserAgent to access this credential.
+
+            # 2. add a password with the -U (update) flag and -w (prompt if last)
+            #    flags, but specify no keychain; security will use the first it finds
+            prepare_darwin_volume_password "$volume" "$new_uuid" | sudo diskutil apfs encryptVolume "$volume" -user disk -stdinpassphrase
```

Comment:
Note that earlier when I recommended we just pass the password in the command-line arguments, that means we can get rid of this two-step keychain setup.

Reply:
Also if we end up keeping the two-step setup we should probably move the first step into `prepare_darwin_volume_password`, it's already responsible for updating the keychain item so it'll be cleaner if we can remove its assumption that the keychain item already exists.

Reply:
Resolving since I've tentatively folded this into `prepare_darwin_volume_password`; two-step handled elsewhere.